sympy/functions/combinatorial/factorials.py

```diff
@@ -1058,31 +1066,34 @@ def _eval_expand_func(self, **hints):
 
     def _eval_rewrite_as_Piecewise(self, n, k, **kwargs):
         from sympy.concrete.products import Product
-        from sympy import gamma
+        from sympy import gamma, Abs, nan
         knk = lambda x: Or(Eq(k, x), Eq(n - k, x))
         isint = lambda x: Eq(x, floor(x))
         nint = lambda x: Ne(x, floor(x))
-        i = Dummy('i', integer=True, positive=True)
+        pos = lambda x: And(Ne(x,0), Eq(x/Abs(x), 1))
+        neg = lambda x: And(Ne(x,0), Eq(x/Abs(x), -1))
+        fin = lambda x: Eq(x/x, nan)
```

Comment:
`fin` looks like `finite` rather than `inf` for `infinite`. I would recommend `inf = lambda x: And(Ne(x, 0), Eq(1/x, 0))`

I feel like Piecewise should have a utility function to write canonical, hardy conditions like this. Also...whitespace after commas (PEP8).

Reply:
Gotcha 👍 , I initially intented for it to be finite, but then switched it up somewhere in the process 😄

Reply:
Let's leave this as unresolved until the change is pushed.